deltachat-contact-tools/src/lib.rs

```diff
@@ -54,6 +54,44 @@ pub struct VcardContact {
     pub timestamp: Result<u64>,
 }
 
+/// Returns a vCard containing given contacts.
+///
+/// Calling [`parse_vcard()`] on the returned result is a reverse operation.
+pub fn make_vcard(contacts: &[VcardContact]) -> String {
+    fn format_timestamp(c: &VcardContact) -> Option<String> {
+        let timestamp = *c.timestamp.as_ref().ok()?;
+        let timestamp: i64 = timestamp.try_into().ok()?;
+        let datetime = DateTime::from_timestamp(timestamp, 0)?;
+        Some(datetime.format("%Y%m%dT%H%M%SZ").to_string())
+    }
+
+    let mut res = "".to_string();
+    for c in contacts {
+        let addr = &c.addr;
+        let display_name = match c.display_name.is_empty() {
+            false => &c.display_name,
+            true => &c.addr,
+        };
+        res += &format!(
+            "BEGIN:VCARD\n\
+             VERSION:4.0\n\
+             EMAIL:{addr}\n\
+             FN:{display_name}\n"
+        );
+        if let Some(key) = &c.key {
+            res += &format!("KEY;PGP;ENCODING=BASE64:{key}\n");
```

Comment:
We're specifying `VERSION:4.0` above; according to [Wikipedia](https://en.wikipedia.org/wiki/VCard) and [the example in the RFC](https://datatracker.ietf.org/doc/html/rfc6350#section-6.8.1) (probably Wikipedia copied from there), this should be `KEY:data:application/pgp-keys;base64,{key}`.

Maybe we should specify `VERSION:2.1`? Or use the 4.0 format? Or is version 4.0 a superset of version 2.1, and the version 2.1 way is still allowed?

Reply:
Thanks, will fix. I think we should generate vCard 4.0 because after googling a bit i see many questions like "[Convert vCard 2.1 to 4.0, how to? : r/ProtonMail](https://www.reddit.com/r/ProtonMail/comments/lrk7ta/convert_vcard_21_to_40_how_to/)". It's quite common to not support older versions